### DISCUSSION TITLE
Only "sun.sun" is supported for area light sensor

### DISCUSSION THREAD

**Discussion by pascalmtts** [jseidl/magic-areas#361] [Q&A] [answered]
Hi there, 
thanks for the update to 4.0.0!
After recreating all creations yesterday I found that the only supported light sensor for controlling the "night" state in an area is sun.sun. Currently I have own binary sensors per room which controlled the night state per room in the previous versions of magic areas. It would be nice to get this feature back, as currently there is no way to control when an area is considered "night" and when not.

What do you think?

**Comment by jseidl** [comment, marked as answer]
It's still there, you just have to set those binary sensors' device class to `light` for them to show up, also the logic is reversed now, they're considered "light" sensors (i.e. the room is NOT dark). 

This is on the release notes
> Goodbye dark entity! Dark entity has been renamed to "Area Light Sensor" as most people used a sort of light binary sensor for that. This change now requires this entity to be a binary sensor and also be of device class light. This helps filtering the select for this entity to only light binary sensors and the sun sensor.

**Reply by pascalmtts** [reply]
Got it thanks for the explanation! 

I think you should add a kind of "Breaking Changes:" to the release notes because I haven't seen this tbh.

**Reply by jseidl** [reply]
You're right, In the rush of getting the release out because 3.x was broken I ended up gobbling all changes together in the notes but indeed should have better announced the breaking changes. Will do better next time!

**Reply by lnlyssg** [reply]
Stupid question but how can I convert an illuminance sensor to a light one? I tried with a template that gives me a true/false based on the illuminance reading but I can't make this a light device class as that isn't supported?

**Reply by TimPasquini** [reply]
> Stupid question but how can I convert an illuminance sensor to a light one? I tried with a template that gives me a true/false based on the illuminance reading but I can't make this a light device class as that isn't supported?

I had a similar problem in that I have some rooms where it's dumb bulbs on a smart switch so I don't have ambient lighting sleep states. I have a master "house_mode" that includes a "night" state. I made an input Boolean called is_night and then an automation that flips it on whenever the house state becomes night and flips it off whenever it leaves night mode. 

You should be able to set up a an automation that takes the illuminance value and flips an input Boolean based on it. 

Once you have a "switch" made you should be able to use the "switch as device" integration to create a helper that converts it to a light entity. That's how I got my smart switches recognized as lights and the bathroom fan.

**Reply by jseidl** [reply]
> Stupid question but how can I convert an illuminance sensor to a light one? 

You need to
1. Use a [Template Binary Sensor](https://www.home-assistant.io/integrations/template/#binary-sensor), something like `{{ states('sensor.kitchen_illuminance')|float > 60 }}`
2. Set this template sensor's device class to `light`

You should be able to select it in Magic Areas. If you're using YAML to create them, you need to set an `unique_id` to be able to change device class and other options.

> I tried with a template that gives me a true/false based on the illuminance reading but I can't make this a light device class as that isn't supported?

Can you clarify?

**Reply by lnlyssg** [reply]
Thanks but I posted this nine months ago and can't recall what I was doing back then!
